napari/layers/image/image.py

```diff
@@ -527,18 +534,46 @@ def interpolation(self):
         str
             The current interpolation mode
         """
-        return str(self._interpolation[self._ndisplay])
+        return str(
+            self._interpolation2d
+            if self._ndisplay == 2
+            else self._interpolation3d
+        )
 
     @interpolation.setter
     def interpolation(self, interpolation):
```

Comment:
`QtImageControls.changeInterpolation` is still using this deprecated setter, but GitHub won't let me comment there. That means that using the GUI control to change the value will always raise a warning, right? I think we need a fix, which isn't just `filterwarnings`.

Reply:
I do not understand why I do not notice warnings when running test.